### PR TITLE
OCPCLOUD-2011: Update implementation history of External platform type EP

### PR DESCRIPTION
**Pull request by adriengentil**
This PR updates the implementation history of the External platform with
the upcoming change in `openshift/api`.

### CODE REVIEW

File: enhancements/cloud-integration/infrastructure-external-platform-type.md [hunk 1]
```diff
@@ -573,6 +573,16 @@ OCP 4.13 will not contain the `CloudControllerManagerSpec` part.
 
 - [PR for removing CloudControllerManagerSpec](https://github.com/openshift/api/pull/1409)
 
+### Add CloudControllerManagerSpec back into the infrastructure resource
+
+Revert the previous desicion to remove CloudControllerManagerSpec from the
```
Comment (bparees) [inline enhancements/cloud-integration/infrastructure-external-platform-type.md:578]:
```suggestion
Revert the previous decision to remove CloudControllerManagerSpec from the
```

**Comment by bparees** [comment]
/approve

thanks for keeping the history updated

**Review by elmiko — COMMENTED** [review]
/lgtm

**Comment by elmiko** [comment]
@JoelSpeed and i discussed this today and i think we've got more questions about the best place to put this field. essentially we have some thoughts about if this should be in the spec or status, and what the most appropriate usage is for users.

/lgtm cancel

**Comment by adriengentil** [comment]
> these updates look nice to me, i'm not sure we want to add any language about how the status will be reconciled. perhaps we should add those details as followup when we get closer to deciding how it will work.

yes, this is what I was thinking, to update this enhancement once we have a clearer picture on how the reconciliation will work.

**Comment by adriengentil** [comment]
As agreed with @elmiko  and @JoelSpeed, I updated the document to control the Cloud Controller Manager state from the status as this setting will be set at installation time and we won't allow it to be changed afterwards.

File: enhancements/cloud-integration/infrastructure-external-platform-type.md [hunk 2]
```diff
@@ -430,12 +402,44 @@ type PlatformSpec struct {
...
+    // state determines whether or not an external Cloud Controller Manager is expected to
+    // be installed within the cluster.
+    // https://kubernetes.io/docs/tasks/administer-cluster/running-cloud-controller/#running-cloud-controller-manager
+    //
+    // When set to "External", new nodes will be tainted as uninitialized when created,
+    // preventing them from running workloads until they are initialized by the cloud controller manager.
+    // When omitted or set to "None", new nodes will be not tainted
+    // and no extra initialization from the cloud controller manager is expected.
+    // +kubebuilder:default:=""
+    // +default=""
```
Comment (JoelSpeed) [inline enhancements/cloud-integration/infrastructure-external-platform-type.md:431]: You don't need to include these I don't think

File: enhancements/cloud-integration/infrastructure-external-platform-type.md [hunk 3]
```diff
@@ -430,12 +402,44 @@ type PlatformSpec struct {
...
+    // +kubebuilder:validation:Enum="";External;None
+    // +optional
+    State CloudControllerManagerState `json:"state"`
```
Comment (JoelSpeed) [inline enhancements/cloud-integration/infrastructure-external-platform-type.md:434]: CEL validation to make this immutable once set will be required

Comment (adriengentil) [reply enhancements/cloud-integration/infrastructure-external-platform-type.md:434]: done

**Comment by adriengentil** [comment]
In my last commit I updated a CEL and mentioned that this new API is behind `TechPreviewNoUpgrade`, so think we should be good.

**Review by elmiko — COMMENTED** [review]
thanks for the updates @adriengentil 

/lgtm

**Pull request merged**